### PR TITLE
Pass the error to inject_listener_closed method

### PR DESCRIPTION
**Pull request by tcharding**
If there is an error when the listener closes, found in the `NetworkEvent::ListenerClosed` `reason` field, we would like to pass it on to the `inject_listener_closed()` method so that implementors of this method have access to it.

Add an error parameter to `inject_listener_closed`.  Convert the `reason` field from a `Result` to an `Option` and if there is an error pass `Some(error)` at the method call site.

Resolves: #1475

### CODE REVIEW

File: swarm/src/behaviour.rs [hunk 1]
```diff
@@ -130,7 +130,7 @@ pub trait NetworkBehaviour: Send + 'static {
     }
 
     /// A listener closed.
-    fn inject_listener_closed(&mut self, _id: ListenerId) {
+    fn inject_listener_closed(&mut self, _id: ListenerId, _err: Option<std::io::Error>) {
```
Comment (tomaka) [inline swarm/src/behaviour.rs]: Could you leave the `Result` for consistency with the rest of the library?

```suggestion
    fn inject_listener_closed(&mut self, _id: ListenerId, reason: Result<(), io::Error>) {
```

Comment (tcharding) [reply swarm/src/behaviour.rs]: Sure thing, can do.

**Comment by tomaka** [comment]
Please pass the error by reference to `inject_listener_closed`

**Comment by tomaka** [comment]
I took the liberty to finish the change, I hope that's ok.
It would be nice to get this PR in the next release, and this next release has already been waiting for a long time.

**Comment by tcharding** [comment]
> Please pass the error by reference to `inject_listener_closed`

Face palm - obviously.  Thanks for finishing it.  FTR feel free to take, modify, paraphrase, or drop anything I contribute.  I'm just here to learn a bit and help out where it helps :)

Thanks.

**Pull request merged**